### PR TITLE
feat: add transaction result subscription

### LINKED ISSUE #9802: Transaction result subscription
Front-end apps have accrued significant technical debt in managing the UI surrounding different transaction types.

Once a transaction is sent to a wallet different subscriptions and logic is needed to properly track the transaction and the resulting changes on the network.

As a result the TransactionResult bus event was added, but it only gets us part of the way there.

This work is to improve that or add a new subscription specifically for this purpose, so that you can easily track transactions _and_ the changes that result from them.

This should also be beneficial in making it easier for external people to build on Vega

### PR DESCRIPTION
**Pull request by karlem**
Close #9802

### CODE REVIEW

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 1]
```diff
@@ -4267,3 +4274,26 @@ message FeesStatsForParty {
   // Total maker fees received by the maker side.
   string total_maker_fees_received = 5;
 }
+
+// Request to subscribe to a stream of user transaction results
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:4278]:
```suggestion
// Request to subscribe to a stream of users' transaction results
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 2]
```diff
@@ -4267,3 +4274,26 @@ message FeesStatsForParty {
...
+  // Request fields party ID, hash and status are all optional filters:
+  // If omitted all transaction results, for all parties with all hashes and states will be returned on the stream
+  // If party ID is given, transaction results from that party will be returned on the stream
+  // If hash is given, transaction result with that hash will be returned on the stream
+  // If status is given, transaction result with that status (true/false - success failure) will be returned on the stream
+  // All filters can be combined.
+
+  // Restrict the transactions results streamed to those made by the given parties.
+  repeated string party_ids = 1;
+  // Restrict the transactions results streamed to those with given hashes.
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:4289]:
```suggestion
  // Restrict the transaction results streamed to those made by the given parties.
  repeated string party_ids = 1;
  // Restrict the transaction results streamed to those with given hashes.
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 3]
```diff
@@ -4267,3 +4274,26 @@ message FeesStatsForParty {
...
+  repeated string hashes = 2;
+  // Restrict the transactions results streamed to those with given status true/false (success/failure).
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:4291]:
```suggestion
  // Restrict the transaction results streamed to those with given status true/false (success/failure).
```

**Comment by wwestgarth** [comment]
I just went to give it a whirl with vegacapsule and got this panic:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10473204c]

goroutine 1907 [running]:
code.vegaprotocol.io/vega/datanode/sqlsubscribers.(*TransactionResults).Observe(0x0, {0x1063139b8, 0x1400016a4b0}, 0x105f3e580?, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, ...)
	/Users/wwestgarth/work/vega/datanode/sqlsubscribers/transaction_results.go:72 +0xdc
code.vegaprotocol.io/vega/datanode/api.(*TradingDataServiceV2).ObserveTransactionResults(0x140019f8500, 0x1400408ec60, {0x1063244c0, 0x1400017b470})
	/Users/wwestgarth/work/vega/datanode/api/trading_data_v2.go:4750 +0xcc
code.vegaprotocol.io/vega/protos/data-node/api/v2._TradingDataService_ObserveTransactionResults_Handler({0x106299b40?, 0x140019f8500}, {0x106321720, 0x140016d8000})
	/Users/wwestgarth/work/vega/protos/data-node/api/v2/trading_data_grpc.pb.go:4963 +0xd4
code.vegaprotocol.io/vega/datanode/gateway.(*SubscriptionRateLimiter).WithGrpcInterceptor.func1({0x106299b40, 0x140019f8500}, {0x106321720?, 0x140016d8000?}, 0x140001fc900, 0x1062c73b0)
	/Users/wwestgarth/work/vega/datanode/gateway/middleware.go:237 +0x2ec
```

**Comment by karlem** [comment]
@wwestgarth thanks for the review. I have fixed it.

**Review by mattrussell36 — DISMISSED** [review]

**Review by wwestgarth — DISMISSED** [review]

**Review by wwestgarth — DISMISSED** [review]

**Review by ValentinTrinque — APPROVED** [review]

**Pull request merged**